lib/Frontend/FrontendOptions.cpp

```diff
@@ -223,11 +223,23 @@ void FrontendOptions::forAllOutputPaths(
   }
   const SupplementaryOutputPaths &outs =
       input.getPrimarySpecificPaths().SupplementaryOutputs;
+  // FIXME: Having to do this mans we need a better way to model supplementary
+  // file paths more abstractly.
   const std::string *outputs[] = {&outs.ModuleOutputPath,
                                   &outs.ModuleDocOutputPath,
                                   &outs.ModuleInterfaceOutputPath,
                                   &outs.PrivateModuleInterfaceOutputPath,
                                   &outs.ObjCHeaderOutputPath,
+                                  &outs.DependenciesFilePath,
```

Comment:
Be careful here—the existing client of `forAllOutputPaths()` is emitting dependency files, and I could imagine that having dependency files refer to themselves might be a problem.

(Perhaps that client can filter on the file type, once you're passing a file type.)